todo/src/components/TaskList.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { addTask } from '../actions'
+import {toggleTask} from '../actions'
+
+class TaskList extends React.Component {
+    state ={
+        newTask: '',
+    }
+
+    handleChanges = e => {
+        this.setState({newTask: e.target.value})
+    }
+
+    addTask = e => {
+        e.preventDefault();
+        this.props.addTask(this.state.newTask)
+        this.setState({newTask: ''})
+    }
+
+    toggleTask = id => {
+        this.props.toggleTask(id)
+    }
+
+
+    render() {
+        return (
+            <>
+                <input type='text'
+                name='newTask'
+                value={this.state.newTask}
+                onChange={this.handleChanges}
+                />
+                
+                <button onClick={this.addTask}>Add Task</button>
```

Comment:
It's usually a better user experience to wrap your input and button in a form, then place your button inside the form and place your submit function(in this case addTask) in an onSubmit on your form tag. This way the user can either press enter or click the button to submit their task.